### ISSUE TITLE
'useRouteMatch' is not exported from 'react-router-dom'

### ISSUE THREAD

**Issue by Quado1** [okta/okta-react#187]
## I'm submitting a:

- [x] Bug report  
- [ ] Feature request
- [ ] Other (Describe below)

## Current behavior

Getting the next error on ```npm start```
```console
Failed to compile.

./node_modules/@okta/okta-react/bundles/okta-react.esm.js
Attempted import error: 'useRouteMatch' is not exported from 'react-router-dom'.
```

Following the [Okta Sign-In Widget and React](https://developer.okta.com/code/react/okta_react_sign-in_widget/) guide

Packages I use:
```json
{
    "@okta/okta-auth-js": "^5.1.1",
    "@okta/okta-react": "^6.0.0",
    "@okta/okta-signin-widget": "^5.7.3",
    "react-router-dom": "^6.0.1",
}
```

The ```okta-react.esm.js``` code parts:
```jsx
import { useRouteMatch, Route } from 'react-router-dom';
...
var match = useRouteMatch(routeProps);
```

## Expected behavior

Compilation without errors

## Minimal reproduction of the problem with instructions
Just follow the guide above

## Extra information about the use case/user story you are trying to implement
I think the problem is the 6th version of the React Router Dom.

UPD: Found the [solution](https://github.com/remix-run/react-router/issues/7133#issuecomment-913009260)

You should use ```useMatch``` in place of ```useRouteMatch```

## Environment

- React version: 17.0.2
- Node version (`node -v`): 16.11.1

## Configuration

- Okta application type: Single Page App (SPA)

**Comment by ryanmr** [comment]
Should we reopen this so that this React Router v6 issue stays in the issues list?

React Router v6 is the new default installation target and this is will block new consumers of this library and react router.

**Comment by Quado1** [comment]
> Should we reopen this so that this React Router v6 issue stays in the issues list?
> 
> React Router v6 is the new default installation target and this is will block new consumers of this library and react router.

There is already [a draft pull request](https://github.com/okta/okta-react/pull/26) where the problem was mentioned

**Comment by ryanmr** [comment]
Oh good catch. It seems like that has been sitting there for a while now.

I hope that gets merged into a new version soon. I'm sure parallel support will be required too.

**Comment by lenichols** [comment]
Has this been merged? I'm running into this same error with update @okta/okta-react

**Comment by jaredperreault-okta** [comment]
@lenichols `@okta/okta-react` currently only supports `react-router` v5. We are currently investigating v6 support

**Comment by johnflux** [comment]
Any progress on this?

**Comment by jaredperreault-okta** [comment]
A fix for this was merged https://github.com/okta/okta-react/pull/188

**Comment by achadha** [comment]
Would that actually fix the issue? You'd still be calling the `useRouteMatch` function which is no longer exported.

**Comment by jaredperreault-okta** [comment]
@achadha you can use `react-router-dom` v6 with `okta-react` if you implement your own `SecureRoute`. More details can be found here https://github.com/okta/okta-react/issues/178

**Comment by flying-sheep** [comment]
I see the same error, in an app not using `<SecureRoute/>` at all.

**Comment by johnflux** [comment]
I found a **really ugly** solution for anyone desperate to just get this working:

In packages.json in your dependencies, do:

```
    "react-router-dom": "npm:react-router-dom@5.3.0",
    "react-router-dom-v6": "npm:react-router-dom@^6.2.1",
```

Then in your own code, import like:

`import { Link } from "react-router-dom-v6";`

**Comment by shuowu** [comment]
@achadha @flying-sheep @johnflux I think the typescript error most probably is introduced from [this type peerDeps](https://github.com/okta/okta-react/blob/master/package.json#L49), which causes type mismatch for react-router-dom v6, we will release a patch for this fix soon.

This issue happens when use npm version that above v7, since peerDeps will be auto installed.

Internal Ref: OKTA-472376

**Comment by shuowu** [comment]
Patch released in [v6.4.3](https://github.com/okta/okta-react/releases/tag/okta-react-6.4.3).

**Comment by baughmann** [comment]
For anyone who's googling this these days:

If you have a dev dependency on `@types/react-router-dom` you're doing it wrong. As of v6 types have been integrated into the lib, so you're including types from v5: https://github.com/nrwl/nx/issues/10238
